src/lightning_app/core/api.py

```diff
@@ -161,86 +167,114 @@ class StateUpdate(BaseModel):
 # Before the above happens, we need to refactor App so that it doesn't
 # rely on timeouts, but on sequences of updates (and alignments between
 # ranks)
-@fastapi_service.get("/api/v1/state", response_class=JSONResponse)
-async def get_state(
-    x_lightning_type: Optional[str] = Header(None),
-    x_lightning_session_uuid: Optional[str] = Header(None),
-    x_lightning_session_id: Optional[str] = Header(None),
-) -> Mapping:
-    if x_lightning_session_uuid is None:
-        raise Exception("Missing X-Lightning-Session-UUID header")
-    if x_lightning_session_id is None:
-        raise Exception("Missing X-Lightning-Session-ID header")
-
+if ENABLE_PULLING_STATE_ENDPOINT:
```

Comment:
Might be better to just raise `HTTP 405 Method Not Allowed` instead of removing the endpoint completely.